src/resolve/resolved-varlink.c

```diff
@@ -37,6 +37,23 @@ static void lookup_parameters_destroy(LookupParameters *p) {
         free(p->name);
 }
 
+static int dns_query_new_for_varlink(
+                Manager *m,
+                DnsQuery **ret,
+                DnsQuestion *question_utf8,
+                DnsQuestion *question_idna,
+                DnsPacket *question_bypass,
+                int ifindex,
+                uint64_t flags) {
+
+        int r = dns_query_new(m, ret, question_utf8, question_idna, question_bypass, ifindex, flags);
+        if (r == -ENOANO) {
+                sd_varlink_error((*ret)->varlink_request, "io.systemd.Resolve.DNSError", NULL);
```

Comment:
No no, `*ret` is `NULL` at this point.
As similar to the DBus case, please add one more argument `sd_varlink *link`, then use it.
Also, propagate error from `sd_varlink_error()`.
```c
return sd_varlink_error(link, "io.systemd.Resolve.DNSError", NULL);
```